### ISSUE TITLE
Student App Recipes

### ISSUE THREAD

**Issue by ds10** [jiscdev/xapi#147]
I think the last recipe we are waiting on for 1.0 is student app. I seem to recall @michaelwebbjisc saying that this was already done and just needed adding to the 1.0 branch?

**Comment by ds10** [comment]
Just bumping this and hoping it lands in the right persons inbox, last call micheal/lee promised to upload them to 1.0 branch?

**Comment by michaelwebbjisc** [comment]
Bump received! will pickup next week

**Comment by michaelwebbjisc** [comment]
I've created a new feature branch feature-study-goal and added:

some generic mobile app templates (eg viewing something on the screen is pretty much the same as viewing a vle page)

https://github.com/jiscdev/xapi/blob/feature-study-goal/recipes/mobile-app.md

some template items for a tracking app (which is what study goal is)

https://github.com/jiscdev/xapi/blob/feature-study-goal/recipes/tracker-app.md

There are a bit rough at the moment but should show what I mean.

**Comment by alanepaull** [comment]
I think the second one is a bit more different from the first one than you've indicated at the moment. The tracker one seems to me to be 'setting a target', and so on, rather than 'viewing a tracker app'. It's an active process rather than simply viewing, because the Actor is changing the data not just looking at a resource.

I suspect that a profitable approach might be to look at a 'targets' recipe, which might apply to a mobile app or VLE (cf. Bedford's Moodle stuff has this, as does the tracker app). This could handle all the student-related aspects of targets - setting the target, updating progress vs target, achieving it, modifying it, achieving late, abandoning it, failing to reach it, and so on. There would be good coherence here in that the Actor could always be the student (though I guess there might be a wrinkle of staff setting a target?), there would be a limited range of 'target-related' verbs (abandoned, commented, completed, created (not sure where this comes from), failed, progressed, viewed), context would be in relation to the particular VLE or other relevant platform, and Object would be the specifics of the target.

Does this help?

**Comment by ds10** [comment]
These have been ssplit but are becoming setting a target app to be introduced in 1.0.2
